src/EventBus/Header.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Patchlevel\EventSourcing\EventBus;
+
+/** @psalm-immutable */
+interface Header
```

Comment:
So i have mixed feelings about this: we dont need that anymore thats true. But what about the psalm immutable annotation? Can we maybe add this information via our psalm plugin (also should do this then for events)? As i think this is needed due to the nature of `Message` being immutable.

Reply:
I'm generally against meta interfaces because an interface should offer something, otherwise you can leave it out.

The attribute `@psalm-immutable` doesn't help either: https://psalm.dev/r/8427a00052

I also don't think that we should educate the user with something like this, but rather write it in the documentation, which explains how you can add a header yourself.

Reply:
The example is more like this: https://psalm.dev/r/290007b891
Still not sure if it would catch something dirty doing by the user. 

I dont think this should be framed as a bad "education". This could lead to bugs if not all methods/calls are pure due e.g. serialization. 

But as i mentioned before, maybe this should be placed then in the psalm plugin, IF we want/need this, since we dont need this interface for anything else.

Reply:
I removed it: as i said, more suitable in plugin i think.